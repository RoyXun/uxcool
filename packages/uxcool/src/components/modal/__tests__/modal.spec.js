import {
  mount,
  waitTime,
  $,
  triggerEvent,
  mountPickerFactory,
  createWrapper,
  destroyWrapper,
} from '@suning/v-test-utils';
import Modal from '../index';

const mountModal = mountPickerFactory(Modal);
describe('modal', () => {
  it('render correctly', async () => {
    const wrapper = await mountModal({
      propsData: {
        value: true,
      },
      slots: {
        title: 'this is title',
      },
    });
    expect($('.ux-modal-close-x').length).toBe(1);
    expect($('.ux-modal-mask').css('display')).toBe('block');
    expect($('.ux-modal-title')[0].innerHTML).toBe('this is title');
    expect(
      $('.ux-modal-footer')[0]
        .children[0].innerHTML.replace(/[\r\n]/, '')
        .trim()
    ).toBe('取消');
    expect(
      $('.ux-modal-footer')[0]
        .children[1].innerHTML.replace(/[\r\n]/, '')
        .trim()
    ).toBe('确定');
    expect($('.ux-modal-footer').find('.ux-btn-primary').length).toBe(1);
    wrapper.setProps({
      closable: false,
      okType: 'danger',
      okText: 'OK',
      cancelText: 'CANCEL',
      zIndex: 999,
      mask: false,
    });
    await waitTime(50);
    expect($('.ux-modal-close-x').length).toBe(0);
    expect($('.ux-modal-mask').length).toBe(0);
    expect(
      $('.ux-modal-footer')[0]
        .children[0].innerHTML.replace(/[\r\n]/, '')
        .trim()
    ).toBe('CANCEL');
    expect(
      $('.ux-modal-footer')[0]
        .children[1].innerHTML.replace(/[\r\n]/, '')
        .trim()
    ).toBe('OK');
    expect($('.ux-modal-footer').find('.ux-btn-danger').length).toBe(1);
    expect($('.ux-modal-wrap').css('z-index')).toBe('999');
    wrapper.setProps({
      hideFooter: true,
    });
    await waitTime(50);
    expect($('.ux-modal-footer').length).toBe(0);
    destroyWrapper(wrapper);
  });

  it('mask closable correctly', async () => {
    const Test = {
      props: {
        closeable: {
          type: Boolean,
          default: false,
        },
      },
      data() {
        return {
          innerVisible: true,
        };
      },
      methods: {
        onInput(visible) {
          this.innerVisible = visible;
        },
      },
      render() {
        const { closeable, innerVisible, onInput } = this;
        const props = {
          value: innerVisible,
          title: 'this is test closable modal',
          maskClosable: closeable,
          transitionName: '',
          maskTransitionName: '',
        };
        const on = {
          input: onInput,
        };
        return (
          <div>
            <Modal {...{ props, on }} />
          </div>
        );
      },
    };
    const wrapper = mount(Test, {
      propsData: {
        closeable: false,
      },
      // attachToDocument: true,
    });

    await waitTime();

    const bodyWrapper = createWrapper(wrapper.element.ownerDocument.body);
    expect(bodyWrapper.find('.ux-modal-wrap').exists()).toBeTruthy();

    await triggerEvent(bodyWrapper.find('.ux-modal-wrap'), 'click');
    expect(bodyWrapper.find('.ux-modal-wrap').element.style.display).not.toBe('none');
    expect(bodyWrapper.classes()).toContain('ux-modal-doc-open');

    // allow mask close
    wrapper.setProps({
      closeable: true,
    });
    await waitTime();

    await triggerEvent(bodyWrapper.find('.ux-modal-wrap'), 'click', 100);
    expect(bodyWrapper.find('.ux-modal-wrap').element.style.display).toBe('none');
    expect(bodyWrapper.classes()).not.toContain('ux-modal-doc-open');
    destroyWrapper(wrapper);
  });

  it('render in custom container correctly', async () => {
    const Test = {
      methods: {
        getPopupContainer() {
          return this.$refs.wrapRef;
        },
      },
      render() {
        const { getPopupContainer } = this;
        return (
          <div ref="wrapRef">
            <Modal value={false} getContainer={getPopupContainer} />
          </div>
        );
      },
    };

    const wrapper = await mount(Test);
    const modalWrapper = wrapper.find(Modal);

    modalWrapper.setProps({ value: true });
    await waitTime();
    expect(wrapper.find('.ux-modal-wrap').exists()).toBeTruthy();
    destroyWrapper(wrapper);
  });

  describe('methods', () => {
    it('render correctly', async () => {
      const okFn = jest.fn();
      const cancelFn = jest.fn();
      Modal.info({
        title: 'this is info title',
        content: '<p>info content</p>',
        dangerouslySetInnerHTML: true,
        okCancel: true,
        cancelText: 'CANCEL',
        okText: 'OK',
        okType: 'danger',
        width: '600px',
        zIndex: 999,
        transitionName: '',
        maskTransitionName: '',
      })
        .then(() => {
          okFn();
        })
        .catch(() => {
          cancelFn();
        });
      await waitTime(100);

      const wrapper = createWrapper(document.querySelector('.ux-modal-light'));
      expect(wrapper.html()).toMatchSnapshot();
      // expect(wrapper.find('.ux-confirm-title').text()).toBe('this is title');
      // expect(wrapper.find('.ux-confirm-content').html()).toBe('<p>content</p>');
      // const btns = wrapper.findAll('.ux-btn');
      // expect(btns.at(0).text()).toBe('CANCEL');
      // expect(btns.at(1).text()).toBe('OK');
      // expect($('.ux-btn-danger').length).toBe(1);
      // expect($('.ux-confirm').css('width')).toBe('600px');
      // expect($('.ux-modal-wrap').css('z-index')).toBe('999');
    });
  });
});
