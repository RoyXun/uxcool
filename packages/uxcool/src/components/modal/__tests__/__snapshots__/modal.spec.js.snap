// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal methods render confirm correctly 1`] = `
<div class="ux-modal-light">
  <div role="mask" class="ux-modal-mask" style="z-index: 999;"></div>
  <div role="dialog" tabindex="-1" class="ux-modal-wrap" style="z-index: 999;">
    <div class="ux-modal ux-confirm ux-confirm-info" style="width: 600px;">
      <div class="ux-modal-content">
        <!---->
        <!---->
        <div class="ux-modal-body">
          <!---->
          <div class="ux-confirm-body-wrapper">
            <div class="ux-confirm-body"><i class="fu fu-info_circle"></i><span class="ux-confirm-title">this is info title</span>
              <div class="ux-confirm-content">
                <p>info content</p>
              </div>
            </div>
            <div class="ux-confirm-btns"><button htmltype="button" class="ux-btn">CANCEL</button><button htmltype="button" class="ux-btn ux-btn-danger">OK</button></div>
          </div>
          <!---->
        </div>
        <!---->
      </div>
    </div>
  </div>
</div>
`;

exports[`modal methods render correctly 1`] = `
<div class="ux-modal-light">
  <div role="mask" class="ux-modal-mask" style="z-index: 999;"></div>
  <div role="dialog" tabindex="-1" class="ux-modal-wrap" style="z-index: 999;">
    <div class="ux-modal ux-confirm ux-confirm-info" style="width: 600px;">
      <div class="ux-modal-content">
        <!---->
        <!---->
        <div class="ux-modal-body">
          <!---->
          <div class="ux-confirm-body-wrapper">
            <div class="ux-confirm-body"><i class="fu fu-info_circle"></i><span class="ux-confirm-title">this is info title</span>
              <div class="ux-confirm-content">
                <p>info content</p>
              </div>
            </div>
            <div class="ux-confirm-btns"><button htmltype="button" class="ux-btn">CANCEL</button><button htmltype="button" class="ux-btn ux-btn-danger">OK</button></div>
          </div>
          <!---->
        </div>
        <!---->
      </div>
    </div>
  </div>
</div>
`;
